### PR TITLE
Java: Use Java 7 try-with-resources for AutoCloseable resources #8213

### LINKED ISSUE #8213: Java: Use Java 7 try-with-resources for AutoCloseable resources
Stuffs like [this](https://github.com/TEAMMATES/teammates/blob/master/src/test/java/teammates/test/driver/FileHelper.java#L23-L32):
```java
BufferedReader br = new BufferedReader(new FileReader(filePath));
Scanner sc = new Scanner(br);
try {
    return sc.useDelimiter("\\Z").next();
} finally {
    sc.close();
    br.close();
}
```
can be modified to use the [try-with-resources statement](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html):
```java
try (Scanner sc = new Scanner(new BufferedReader(new FileReader(filePath)))) {
    return sc.useDelimiter("\\Z").next();
}
```
The benefit is two-fold: makes the code shorter and also closes the linked resources without fail, although admittedly I cannot say for sure what will happen if the resources do leak.

There are more than 10 places in the code base where this can be applied. Some common [`AutoCloseable`](https://docs.oracle.com/javase/8/docs/api/java/lang/AutoCloseable.html) resources are `FileWriter`, `BufferedWriter`, `PrintWriter`, `FileOutputStream`, `OutputStreamWriter`, `Scanner`, `FileReader`, `BufferedReader`, `FileInputStream`, `InputStreamReader`.

### PR DESCRIPTION
**Pull request by upatidar7**
Part of #8213 

removed all try with try-with-resources .
Ready for review.

please let me know if you want any changes, it's my first contribution to open source. 
Thanks.

### CODE REVIEW

**Comment by upatidar7** [comment]
waiting for your response.

**Comment by wkurniawan07** [comment]
@upatidar7 please make sure that all checks have passed first. Right now Travis CI still gives a red "X".

**Comment by upatidar7** [comment]
how to remove all error which is showing in checkstyle , is there any shortcut to fix all error in intellij ..?

**Comment by upatidar7** [comment]
i am getting an error in checkstyle 
"The Checkstyle suppression filter file was not found and will be ignored"

**Comment by wkurniawan07** [comment]
> how to remove all error which is showing in checkstyle , is there any shortcut to fix all error in intellij

@upatidar7 Not that I know of, but try not to be lazy. The things you need to do are clearly listed in the Travis build log.

> The Checkstyle suppression filter file was not found and will be ignored

You can run Checkstyle outside of IDEs with `./gradlew checkstyleMain` or `./gradlew checkstyleTest` (`gradlew.bat *` in Windows).

**Comment by upatidar7** [comment]
@wkurniawan07 I am getting following error in logs and I don't know where is it failing..? 


npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @ lint:spaces: `lintspaces -n -t -d spaces -l 1 -. "src/**/*.jsp" "src/**/*.tag" "src/main/**/*.html" "src/**/*.xml" "src/**/*.json" "src/**/*.css" "src/**/*.java" "src/main/webapp/dev/**/*.js" "src/**/*.properties" "*.js" "*.yml" "*.gradle" "static-analysis/*.*ml"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @ lint:spaces script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2018-01-16T16_54_28_743Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @ lint: `npm run lint:js && npm run lint:json && npm run lint:css && npm run lint:spaces`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @ lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2018-01-16T16_54_28_754Z-debug.log
travis_
The command "npm run lint" failed and exited with 1 during .

**Comment by wkurniawan07** [comment]
@upatidar7 take a look at a few lines just before that.

**Comment by upatidar7** [comment]
where I can find following logs -
" /home/travis/.npm/_logs/2018-01-17T15_08_52_314Z-debug.log"

**Comment by wkurniawan07** [comment]
@upatidar7 you do not need that log file (nor you can access it, also). Just look at the Travis build log a few lines before the block of `npm ERR!`s.

**Comment by upatidar7** [comment]
@wkurniawan07 I have cleared all checks.

File: src/client/java/teammates/client/scripts/ModifyInstituteOfStudentsInCourse.java [hunk 1]
```diff
@@ -19,14 +19,13 @@ public static void main(String[] args) throws IOException {
...
-        String courseId = scanner.nextLine();
-        System.out.println("Enter new institute name: ");
-        String institute = scanner.nextLine();
+        try (Scanner scanner = new Scanner(System.in)) {
+            System.out.println("Enter course to edit: ");
+            String courseId = scanner.nextLine();
+            System.out.println("Enter new institute name: ");
+            String institute = scanner.nextLine();
 
-        try {
```
[the remark this answers is not in the document]
Comment (upatidar7) [reply src/client/java/teammates/client/scripts/ModifyInstituteOfStudentsInCourse.java]: done

**Comment by upatidar7** [comment]
@wkurniawan07 done

File: src/client/java/teammates/client/scripts/GenerateFeedbackReport.java [hunk 2]
```diff
@@ -27,7 +27,7 @@ protected void doOperation() {
         try {
             String fileContent =
                     logic.getFeedbackSessionResultSummaryAsCsv(
-                            "CourseID", "Session Name", "instructor@email.com", true, true, null);
+                    "CourseID", "Session Name", "instructor@email.com", true, true, null);
```
[the remark this answers is not in the document]
Comment (upatidar7) [reply src/client/java/teammates/client/scripts/GenerateFeedbackReport.java]: Tried but showing error in check style that incorrect indentation level

**Comment by upatidar7** [comment]
I used "find in path in IntelliJ" which is scanning java files, XML files, JSON files and HTML files, and comments also, and pretty sure that  there is no place other than this  for "try with resource"

File: src/client/java/teammates/client/scripts/GenerateFeedbackReport.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (upatidar7) [reply src/client/java/teammates/client/scripts/GenerateFeedbackReport.java]: `:lintMain
:compileTestJava UP-TO-DATE
:processTestResources UP-TO-DATE
:testClasses UP-TO-DATE
:checkstyleTest[ant:checkstyle] [ERROR] /home/travis/build/TEAMMATES/teammates/src/client/java/teammates/client/scripts/GenerateFeedbackReport.java:30: '"CourseID"' has incorrect indentation level 12, expected level should be 20. [Indentation]
 FAILED
:findbugsTest
:pmdTest
:macker`

Have a look ,  not passing in checkstyle.

[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/client/java/teammates/client/scripts/GenerateFeedbackReport.java]: @upatidar7 do you realize that you're not reverting it to the old state?

[the remark this answers is not in the document]
Comment (upatidar7) [reply src/client/java/teammates/client/scripts/GenerateFeedbackReport.java]: sorry , but I am not getting. pls explain what is going on ?

[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/client/java/teammates/client/scripts/GenerateFeedbackReport.java]: @upatidar7 you're not fixing this; in fact [making it worse](https://github.com/TEAMMATES/teammates/pull/8308/files#diff-91c83125cd33d8e866d183c1f77aee5eR30).

**Comment by upatidar7** [comment]
Now it is showing all checks passes...?

File: src/client/java/teammates/client/scripts/GenerateFeedbackReport.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (upatidar7) [reply src/client/java/teammates/client/scripts/GenerateFeedbackReport.java]: Sorry my bad, there is a problem with IntelliJ.

**Comment by wkurniawan07** [comment]
> I used "find in path in IntelliJ"

@upatidar7 I'm not sure what you're searching for but [this](https://github.com/TEAMMATES/teammates/blob/master/src/test/java/teammates/test/driver/GmailServiceMaker.java#L90-L97) and [this](https://github.com/TEAMMATES/teammates/blob/master/src/client/java/teammates/client/remoteapi/RemoteApiClient.java#L45-L52) are examples of where try-with-resources can be used.

**Comment by upatidar7** [comment]
thanks

**Comment by whipermr5** [comment]
@upatidar7 Thanks for your contribution! May we have your full name for inclusion into the contributors' list? :)

**Comment by upatidar7** [comment]
Why not,  Umesh Patidar.

**Pull request merged**